dockernet/ts-tests/test/main.test.ts

```diff
@@ -0,0 +1,180 @@
+
+const RPC_ENDPOINT = "http://localhost:26657";
+
+let accounts: {
+  user: StrideClient; // just a normal user account loaded with 100 STRD
+  admin: StrideClient; // the stride admin account loaded with 1000 STRD
+  val1: StrideClient;
+  val2: StrideClient;
+  val3: StrideClient;
+};
+
+// init accounts and wait for chain to start
+beforeAll(async () => {
+  console.log("setting up accounts...");
+
+  const mnemonics: {
+    name: "user" | "admin" | "val1" | "val2" | "val3";
+    mnemonic: string;
+  }[] = [
+    {
+      name: "user",
+      mnemonic:
+        "brief play describe burden half aim soccer carbon hope wait output play vacuum joke energy crucial output mimic cruise brother document rail anger leaf",
+    },
+    {
+      name: "admin",
+      mnemonic:
+        "tone cause tribe this switch near host damage idle fragile antique tail soda alien depth write wool they rapid unfold body scan pledge soft",
+    },
+    {
+      name: "val1",
+      mnemonic:
+        "close soup mirror crew erode defy knock trigger gather eyebrow tent farm gym gloom base lemon sleep weekend rich forget diagram hurt prize fly",
+    },
+    {
+      name: "val2",
+      mnemonic:
+        "turkey miss hurry unable embark hospital kangaroo nuclear outside term toy fall buffalo book opinion such moral meadow wing olive camp sad metal banner",
+    },
+    {
+      name: "val3",
+      mnemonic:
+        "tenant neck ask season exist hill churn rice convince shock modify evidence armor track army street stay light program harvest now settle feed wheat",
+    },
+  ];
+
+  // @ts-expect-error
+  // init accounts as an empty object, then add the accounts in the loop
+  accounts = {};
+  for (const { name, mnemonic } of mnemonics) {
+    // @ts-expect-error
+    // init accounts[name] as an empty object, then add the fields one by one
+    accounts[name] = {};
+
+    // setup signer
+    //
+    // IMPORTANT: we're using Secp256k1HdWallet from @cosmjs/amino because sending amino txs tests both amino and direct.
+    // that's because the tx contains the direct encoding anyway, and also attaches a signature on the amino encoding.
+    // the mempool then converts from direct to amino to verify the signature.
+    // therefore if the signature verification passes, we can be sure that both amino and direct are supported.
+    const signer = await Secp256k1HdWallet.fromMnemonic(mnemonic, {
+      prefix: "stride",
+    });
+
+    // get signer address
+    const [{ address }] = await signer.getAccounts();
+
+    accounts[name] = await StrideClient.create(RPC_ENDPOINT, signer, address, {
+      gasPrice: GasPrice.fromString("0.025ustrd"),
+      broadcastPollIntervalMs: 50,
+      resolveIbcResponsesCheckIntervalMs: 50,
+    });
+  }
+
+  console.log("waiting for chain to start...");
+  while (true) {
+    try {
+      const block =
+        await accounts.user.query.cosmos.base.tendermint.v1beta1.getLatestBlock(
```

Comment:
do you have to execute all queries through a user like this?

Reply:
yes though it's the same query client. we can export it to a global query variable for these tests, but in general I've learned from secret that it's more convenient to have the query and tx clients under the same object

Reply:
open to more feedback though

Reply:
Gotchya - let's keep it for now and we can iterate on it!

Fwiw, the global query option sounds like a good option. I setup [firehose](https://github.com/Stride-Labs/firehose) with the query/tx client on the user and always felt like it was a bit of an antipattern

Reply:
just curious - what's the benefit you noticed from secret in having it be shared?

Reply:
you just have the one StrideClient object, you can traverse it just like the cli
stridejs.query.bank.balance...
stridejs.tx.bank.send...

we've got a lot of good feedback on this after multiple iterations, but mostly that it's nice that you can explore the object with the ide's intelisense like you would explore the cli with typing half a command and appendig --help

Reply:
nice, make sense! I do think it breaks down a bit though when you have multiple accounts that you're sending txs from in one test. In which case, I could imagine it being cleaner to have a separate global query config

```
query.bank...
accounts.X.tx.bank.send(...)
accounts.Y.tx.bank.send(...)
```

But just ideating - obviously no need to change anything now!

Reply:
I agree